drivers/bluetooth/hci/cyw43xxx.c

```diff
@@ -0,0 +1,300 @@
+/*
+ * Copyright (c) 2022 Cypress Semiconductor Corporation (an Infineon company) or
+ * an affiliate of Cypress Semiconductor Corporation
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+/**
+ * @brief CYW43xxx HCI extension driver.
+ */
+
+#include <errno.h>
+#include <stddef.h>
+
+#include <zephyr/kernel.h>
+#include <zephyr/device.h>
+#include <zephyr/bluetooth/bluetooth.h>
+#include <zephyr/drivers/bluetooth/hci_driver.h>
+#include <zephyr/drivers/gpio.h>
+#include <zephyr/drivers/uart.h>
+
+#define BT_DBG_ENABLED IS_ENABLED(CONFIG_BT_DEBUG_HCI_DRIVER)
+#define LOG_LEVEL CONFIG_BT_HCI_DRIVER_LOG_LEVEL
+#include <zephyr/logging/log.h>
+LOG_MODULE_REGISTER(cyw43xxx_driver);
+
+#include <stdint.h>
+
+#define HCI_UART_NODE                     DT_COMPAT_GET_ANY_STATUS_OKAY(infineon_cyw43xxx_bt_hci)
+
+/* BT settling time after power on */
+#define BT_POWER_ON_SETTLING_TIME         (500u)
+
+/* Stabilization delay after FW loading */
+#define BT_STABILIZATION_DELAY            (250)
+
+/* HCI Command packet from Host to Controller */
+#define HCI_COMMAND_PACKET                (0x01)
+
+/* Length of UPDATE BAUD RATE command */
+#define HCI_VSC_UPDATE_BAUD_RATE_LENGTH   (6u)
+
+/* Default BAUDRATE */
+#define HCI_UART_DEFAULT_BAUDRATE         (115200)
```

Comment:
What if the BT controller will use another baudrate by default? Would be good to have that configurable.

Another thing is the baudrate switch logic. As part of https://github.com/zephyrproject-rtos/zephyr/pull/28744 there is following behavior:
- use `current-speed` of UART bus node by default, for initial transmission,
- switch to `target-speed` of ESP-AT node (which is a child of UART node) by "negotiating" speed in runtime.

Your approach is inverted compared to ESP-AT driver:
- use hardcoded `HCI_UART_DEFAULT_BAUDRATE` speed by default for initial transmission,
- switch to `current-speed` of UART bus node by "negotiating" speed in runtime.

Would be good to have consistent behavior in that regard. IMO we should use one baudrate by default (set `current-speed` of UART node to the one this BT controller will use by default) and provide options to use custom baudrates for firmware transfer and/or further communication (after initialization).

Reply:
Yes, that is exactly what I have written above in:
> Your approach is inverted compared to ESP-AT driver:
>
>use hardcoded HCI_UART_DEFAULT_BAUDRATE speed by default for initial transmission,
>switch to current-speed of UART bus node by "negotiating" speed in runtime.

but there is a switch to `fw-download-speed` between those two steps. I am aware of that (additional) step, but the relevant part is that this is different from what ESP-AT driver does and how it respects `current-speed` of the bus.

Reply:
let me check ESP-AT driver...

Reply:
After power ON, or after FW downloading complete, CYW43xxx must start HCI UART communication on 115200 (HCI_UART_DEFAULT_BAUDRATE)  So in CYW43xxx  it does not make sense  to use HCI_UART_DEFAULT_BAUDRATE as `current-speed`. That why, i am using:
-  `current-speed` as baudrate for common HCI UART communication. And we switch to this speed after fw downloading.
- `fw-download-speed` - specific baudrate configuration which we used during fw downloading stage.

Let me know if it is clear.

Reply:
> After power ON, or after FW downloading complete, CYW43xxx must start HCI UART communication on 115200

Is this true for every CYW43xxx available on the market?

I still think that 115200 should not be hardcoded and should come from devicetree. This is true for most (all?) other device drivers. Looking at the devicetree alone allows to tell what will be the UART speed used during communication. `espressif,esp-at` and `swir,hl7800` are some of the examples.

Another reason to have 115200 in devicetree (from UART bus `current-speed` property specifically) is that this speed is configured automatically in serial device drivers, without the need to reconfigure in runtime. We even have `CONFIG_UART_USE_RUNTIME_CONFIGURE` which can be disabled, so any changes in runtime might not be supported. In that case this default speed (115200 configured in UART `current-speed`) should be used for initialization, FW dowload and operation.

So what I try to suggest from the beginning is to use dedicated property for "operation speed". This would be aligned with ESP-AT driver as well and make both consistent.

Reply:
updated.